### PR TITLE
Graph optimization

### PR DESCRIPTION
**Pull request by soujay**
This PR includes the following graph optimizations.
- fix arrow label pointer
- turn off eslint check for unused vars
- remove redundant properties of graph properties

### CODE REVIEW

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue [hunk 1]
```diff
@@ -196,7 +196,7 @@ function formatGraphDecimal(input, deci) {
                 <div class="box-header">
                     <div class="no-side-padding">
                         <!--<EndpointGraph :type="'throughput'"></EndpointGraph>-->
-                        <D3Graph :endpointname="endpoint.name" :colname="'throughput'" :isdurationgraph="false" :plotdata="endpoint.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'" :csclass="'graph throughput pull-left ng-isolate-scope'"></D3Graph>
+                        <D3Graph  :type="'throughput'" :isdurationgraph="false" :plotdata="endpoint.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'" :csclass="'graph throughput pull-left ng-isolate-scope'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue:199]:
```suggestion
                        <D3Graph  :type="'throughput'" :isdurationgraph="false" :plotdata="endpoint.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue [hunk 2]
```diff
@@ -182,7 +182,7 @@ function formatGraphDecimal(input, deci) {
                 <div class="box-header">
                     <div class="no-side-padding">
                         <!--<EndpointGraph :type="'queue-length'"></EndpointGraph>-->
-                        <D3Graph :endpointname="endpoint.name" :colname="'queuelength'" :isdurationgraph="false" :plotdata="endpoint.metrics.queueLength" :minimumyaxis="smallGraphsMinimumYAxis.queueLength" :avglabelcolor="'#EA7E00'" :metricsuffix="'MSGS'" :csclass="'graph queue-length pull-left ng-isolate-scope'"></D3Graph>
+                        <D3Graph :type="'queue-length'" :isdurationgraph="false" :plotdata="endpoint.metrics.queueLength" :minimumyaxis="smallGraphsMinimumYAxis.queueLength" :avglabelcolor="'#EA7E00'" :metricsuffix="'MSGS'" :csclass="'graph queue-length pull-left ng-isolate-scope'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue:185]:
```suggestion
                        <D3Graph :type="'queue-length'" :isdurationgraph="false" :plotdata="endpoint.metrics.queueLength" :minimumyaxis="smallGraphsMinimumYAxis.queueLength" :avglabelcolor="'#EA7E00'" :metricsuffix="'MSGS'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue [hunk 3]
```diff
@@ -210,7 +210,7 @@ function formatGraphDecimal(input, deci) {
                 <div class="box-header">
                     <div class="no-side-padding">
                         <!--<EndpointGraph :type="'retries'"></EndpointGraph>-->
-                        <D3Graph :endpointname="endpoint.name" :colname="'retries'" :isdurationgraph="false" :plotdata="endpoint.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'" :csclass="'graph retries pull-left ng-isolate-scope'"></D3Graph>
+                        <D3Graph :type="'retries'" :isdurationgraph="false" :plotdata="endpoint.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'" :csclass="'graph retries pull-left ng-isolate-scope'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue:213]:
```suggestion
                        <D3Graph :type="'retries'" :isdurationgraph="false" :plotdata="endpoint.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue [hunk 4]
```diff
@@ -224,7 +224,7 @@ function formatGraphDecimal(input, deci) {
                 <div class="box-header">
                     <div class="no-side-padding">
                         <!--<EndpointGraph :type="'processing-time'"></EndpointGraph>-->
-                        <D3Graph :endpointname="endpoint.name" :colname="'processingtime'" :isdurationgraph="true" :plotdata="endpoint.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'" :csclass="'graph processing-time pull-left ng-isolate-scope'"></D3Graph>
+                        <D3Graph  :type="'processing-time'" :isdurationgraph="true" :plotdata="endpoint.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'" :csclass="'graph processing-time pull-left ng-isolate-scope'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue:227]:
```suggestion
                        <D3Graph  :type="'processing-time'" :isdurationgraph="true" :plotdata="endpoint.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue [hunk 5]
```diff
@@ -239,7 +239,7 @@ function formatGraphDecimal(input, deci) {
                 <div class="box-header">
                     <div class="no-side-padding">
                         <!--<EndpointGraph :type="'critical-time'"></EndpointGraph>-->
-                        <D3Graph :endpointname="endpoint.name" :colname="'criticaltime'" :isdurationgraph="true" :plotdata="endpoint.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'" :csclass="'graph critical-time pull-left ng-isolate-scope'"></D3Graph>
+                        <D3Graph  :type="'critical-time'" :isdurationgraph="true" :plotdata="endpoint.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'" :csclass="'graph critical-time pull-left ng-isolate-scope'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue:242]:
```suggestion
                        <D3Graph  :type="'critical-time'" :isdurationgraph="true" :plotdata="endpoint.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 6]
```diff
@@ -579,8 +574,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                               <div class="no-side-padding">
-                                <!--<graph plot-data="instance.metrics.throughput" minimum-YAxis="{{smallGraphsMinimumYAxis.throughput}}" avg-label-color="#176397" metric-suffix="MSGS/S" class="graph throughput pull-left"></graph>-->
-                                <D3Graph :endpointname="endpoint.name" :colname="'throughput'" :isdurationgraph="false" :plotdata="instance.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'" :csclass="'graph throughput pull-left'"></D3Graph>
+                                <D3Graph  :type="'throughput'" :isdurationgraph="false" :plotdata="instance.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'" :csclass="'graph throughput pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:577]:
```suggestion
                                <D3Graph  :type="'throughput'" :isdurationgraph="false" :plotdata="instance.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 7]
```diff
@@ -592,8 +586,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                                 <div class="no-side-padding">
-                                    <!--<graph plot-data="instance.metrics.retries" minimum-YAxis="{{smallGraphsMinimumYAxis.retries}}" avg-label-color="#CC1252" metric-suffix="MSGS/S" class="graph retries pull-left"></graph>-->
-                                    <D3Graph :endpointname="endpoint.name" :colname="'retries'" :isdurationgraph="false" :plotdata="instance.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'" :csclass="'graph retries pull-left'"></D3Graph>
+                                    <D3Graph  :type="'retries'" :isdurationgraph="false" :plotdata="instance.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'" :csclass="'graph retries pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:589]:
```suggestion
                                    <D3Graph  :type="'retries'" :isdurationgraph="false" :plotdata="instance.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 8]
```diff
@@ -605,8 +598,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                               <div class="no-side-padding">
-                                <!--<graph plot-data="instance.metrics.processingTime" minimum-YAxis="{{smallGraphsMinimumYAxis.processingTime}}" avg-label-color="#258135" is-duration-graph="true" class="graph processing-time pull-left"></graph>-->
-                                <D3Graph :endpointname="endpoint.name" :colname="'processingtime'" :isdurationgraph="true" :plotdata="instance.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'" :csclass="'graph processing-time pull-left'"></D3Graph>
+                                <D3Graph  :type="'processing-time'" :isdurationgraph="true" :plotdata="instance.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'" :csclass="'graph processing-time pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:601]:
```suggestion
                                <D3Graph  :type="'processing-time'" :isdurationgraph="true" :plotdata="instance.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 9]
```diff
@@ -621,8 +613,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                               <div class="no-side-padding">
-                                <!--<graph plot-data="instance.metrics.criticalTime" minimum-YAxis="{{smallGraphsMinimumYAxis.criticalTime}}" avg-label-color="#2700CB" is-duration-graph="true" class="graph critical-time pull-left"></graph>-->
-                                <D3Graph :endpointname="endpoint.name" :colname="'criticaltime'" :isdurationgraph="true" :plotdata="instance.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'" :csclass="'graph critical-time pull-left'"></D3Graph>
+                                <D3Graph  :type="'critical-time'" :isdurationgraph="true" :plotdata="instance.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'" :csclass="'graph critical-time pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:616]:
```suggestion
                                <D3Graph  :type="'critical-time'" :isdurationgraph="true" :plotdata="instance.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 10]
```diff
@@ -723,8 +714,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                               <div class="no-side-padding">
-                                <!--<graph plot-data="messageType.metrics.throughput" minimum-YAxis="{{smallGraphsMinimumYAxis.throughput}}" avg-label-color="#176397" metric-suffix="MSGS/S" class="graph throughput pull-left"></graph>-->
-                                <D3Graph :endpointname="endpoint.name" :colname="'throughput'" :isdurationgraph="false" :plotdata="messageType.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'" :csclass="'graph throughput pull-left'"></D3Graph>
+                                <D3Graph  :type="'throughput'" :isdurationgraph="false" :plotdata="messageType.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'" :csclass="'graph throughput pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:717]:
```suggestion
                                <D3Graph  :type="'throughput'" :isdurationgraph="false" :plotdata="messageType.metrics.throughput" :minimumyaxis="smallGraphsMinimumYAxis.throughput" :avglabelcolor="'#176397'" :metricsuffix="'MSGS/S'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 11]
```diff
@@ -736,8 +726,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                                 <div class="no-side-padding">
-                                    <!--<graph plot-data="messageType.metrics.retries" minimum-YAxis="{{smallGraphsMinimumYAxis.retries}}" avg-label-color="#CC1252" metric-suffix="MSGS/S" class="graph retries pull-left"></graph>-->
-                                    <D3Graph :endpointname="endpoint.name" :colname="'retries'" :isdurationgraph="false" :plotdata="messageType.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'" :csclass="'graph retries pull-left'"></D3Graph>
+                                    <D3Graph  :type="'retries'" :isdurationgraph="false" :plotdata="messageType.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'" :csclass="'graph retries pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:729]:
```suggestion
                                    <D3Graph  :type="'retries'" :isdurationgraph="false" :plotdata="messageType.metrics.retries" :minimumyaxis="smallGraphsMinimumYAxis.retries" :avglabelcolor="'#CC1252'" :metricsuffix="'MSGS/S'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 12]
```diff
@@ -749,8 +738,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                                 <div class="no-side-padding">
-                                    <!--<graph plot-data="messageType.metrics.processingTime" minimum-YAxis="{{smallGraphsMinimumYAxis.processingTime}}" avg-label-color="#258135" is-duration-graph="true" class="graph processing-time pull-left"></graph>-->
-                                    <D3Graph :endpointname="endpoint.name" :colname="'processingtime'" :isdurationgraph="true" :plotdata="messageType.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'" :csclass="'graph processing-time pull-left'"></D3Graph>
+                                    <D3Graph  :type="'processing-time'" :isdurationgraph="true" :plotdata="messageType.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'" :csclass="'graph processing-time pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:741]:
```suggestion
                                    <D3Graph  :type="'processing-time'" :isdurationgraph="true" :plotdata="messageType.metrics.processingTime" :minimumyaxis="smallGraphsMinimumYAxis.processingTime" :avglabelcolor="'#258135'"></D3Graph>
```

File: src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue [hunk 13]
```diff
@@ -765,8 +753,7 @@ onMounted(() => {
                           <div class="col-xs-2 col-xl-1 no-side-padding">
                             <div class="row box-header">
                               <div class="no-side-padding">
-                                <!--<graph plot-data="messageType.metrics.criticalTime" minimum-YAxis="{{smallGraphsMinimumYAxis.criticalTime}}" avg-label-color="#2700CB" is-duration-graph="true" class="graph critical-time pull-left"></graph>-->
-                                 <D3Graph :endpointname="endpoint.name" :colname="'criticaltime'" :isdurationgraph="true" :plotdata="messageType.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'" :csclass="'graph critical-time pull-left'"></D3Graph>
+                                 <D3Graph  :type="'critical-time'" :isdurationgraph="true" :plotdata="messageType.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'" :csclass="'graph critical-time pull-left'"></D3Graph>
```
Comment (cquirosj) [inline src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue:756]:
```suggestion
                                 <D3Graph  :type="'critical-time'" :isdurationgraph="true" :plotdata="messageType.metrics.criticalTime" :minimumyaxis="smallGraphsMinimumYAxis.criticalTime" :avglabelcolor="'#2700CB'"></D3Graph>
```

**Comment by cquirosj** [comment]
@soujay could we create stories showcasing the different usage combinations? Some examples that come to mind (just ideas not, don't take them as mandatory):
- Graph with one series
- Graph with two series
- Flat graph

Could we also write [interactions](https://storybook.js.org/docs/vue/writing-stories/play-function) so that the following is checked:
- That the average lines got rendered
- That the [expected computed](https://github.com/Particular/ServicePulse/blob/c00377738e32e0be991c375445f25b63ad50899f/src/ServicePulse.Host/vue/src/components/monitoring/EndpointGraph.stories.js#L21) colors for a given 'type' are correct. 
- That on [mouse hover](https://testing-library.com/docs/ecosystem-user-event/#hoverelement-options) the arrow labels get displayed and when the mouse exists the arrows hide?

Could we also remove the static component and stories for EndpointGraph.vue

**Comment by cquirosj** [comment]
> @soujay could we create stories showcasing the different usage combinations? Some examples that come to mind (just ideas not, don't take them as mandatory):
> 
>     * Graph with one series
> 
>     * Graph with two series
> 
>     * Flat graph
> 
> 
> Could we also write [interactions](https://storybook.js.org/docs/vue/writing-stories/play-function) so that the following is checked:
> 
>     * That the average lines got rendered
> 
>     * That the [expected computed](https://github.com/Particular/ServicePulse/blob/c00377738e32e0be991c375445f25b63ad50899f/src/ServicePulse.Host/vue/src/components/monitoring/EndpointGraph.stories.js#L21) colors for a given 'type' are correct.
> 
>     * That on [mouse hover](https://testing-library.com/docs/ecosystem-user-event/#hoverelement-options) the arrow labels get displayed and when the mouse exists the arrows hide?
> 
> 
> Could we also remove the static component and stories for EndpointGraph.vue

If you agree with all these ideas, we can also work on that on a separate PR if we prefer.

**Comment by soujay** [comment]
@cquirosj  - Yes, I will add in more stories for the graph component in a separate PR. I have removed the  removed the static component and stories for EndpointGraph.vue as part of this PR

**Comment by cquirosj** [comment]
Looks like the stories for D3Graph still make use of [colname](https://github.com/Particular/ServicePulse/blob/7194ba3a22863d3493d28724565b4198bf0b78d1/src/ServicePulse.Host/vue/src/components/monitoring/D3Graph.stories.js#L25) and [csClass](https://github.com/Particular/ServicePulse/blob/7194ba3a22863d3493d28724565b4198bf0b78d1/src/ServicePulse.Host/vue/src/components/monitoring/D3Graph.stories.js#L23), we still need to do more focused work so that the stories really behave as tests making assertions rather than only rendering the components, this way, if we remove something from the components, the tests will start failing. Once we work on separte PRs to address this we can do the clean-up of the no longer used args colname and csClass

**Review by cquirosj — APPROVED** [review]

**Pull request merged**